spring-boot-starter-camunda/src/main/java/io/camunda/zeebe/spring/client/configuration/ZeebeClientConfiguration.java

```diff
@@ -0,0 +1,198 @@
+    return properties.getNumJobWorkerExecutionThreads();
+  }
+
+  @Override
+  public int getDefaultJobWorkerMaxJobsActive() {
+    return properties.getDefaultJobWorkerMaxJobsActive();
+  }
+
+  @Override
+  public String getDefaultJobWorkerName() {
+    return properties.getDefaultJobWorkerName();
+  }
+
+  @Override
+  public Duration getDefaultJobTimeout() {
+    return properties.getDefaultJobTimeout();
+  }
+
+  @Override
+  public Duration getDefaultJobPollInterval() {
+    return properties.getDefaultJobPollInterval();
+  }
+
+  @Override
+  public Duration getDefaultMessageTimeToLive() {
+    return properties.getDefaultMessageTimeToLive();
+  }
+
+  @Override
+  public Duration getDefaultRequestTimeout() {
+    return properties.getDefaultRequestTimeout();
+  }
+
+  @Override
+  public boolean isPlaintextConnectionEnabled() {
+    return properties.isPlaintextConnectionEnabled();
+  }
+
+  @Override
+  public String getCaCertificatePath() {
+    return properties.getCaCertificatePath();
+  }
+
+  @Override
+  public CredentialsProvider getCredentialsProvider() {
+    if (!(authentication instanceof DefaultNoopAuthentication)) {
+      return new CredentialsProvider() {
+        @Override
+        public void applyCredentials(Metadata headers) throws IOException {
+          final var authHeader = authentication.getTokenHeader(Product.ZEEBE);
+          final var authHeaderKey = Metadata.Key.of(authHeader.getKey(), Metadata.ASCII_STRING_MARSHALLER);
+          headers.put(authHeaderKey, authHeader.getValue());
+        }
+
+        @Override
+        public boolean shouldRetryRequest(Throwable throwable) {
+          return ((StatusRuntimeException) throwable).getStatus() == Status.DEADLINE_EXCEEDED;
+        }
+      };
+    }
+    if (hasText(properties.getCloud().getClientId()) && hasText(properties.getCloud().getClientSecret())) {
+//        log.debug("Client ID and secret are configured. Creating OAuthCredientialsProvider with: {}", this);
+      return CredentialsProvider.newCredentialsProviderBuilder()
+        .clientId(properties.getCloud().getClientId())
+        .clientSecret(properties.getCloud().getClientSecret())
+        .audience(properties.getCloud().getAudience())
+        .scope(properties.getCloud().getScope())
+        .authorizationServerUrl(properties.getCloud().getAuthUrl())
+        .credentialsCachePath(properties.getCloud().getCredentialsCachePath())
+        .build();
+    }
+    if (Environment.system().get("ZEEBE_CLIENT_ID") != null && Environment.system().get("ZEEBE_CLIENT_SECRET") != null) {
+      // Copied from ZeebeClientBuilderImpl
+      OAuthCredentialsProviderBuilder builder = CredentialsProvider.newCredentialsProviderBuilder();
+      int separatorIndex = properties.getBroker().getGatewayAddress().lastIndexOf(58); //":"
+      if (separatorIndex > 0) {
+        builder.audience(properties.getBroker().getGatewayAddress().substring(0, separatorIndex));
+      }
+      return builder.build();
+    }
```

Comment:
You may refactor this as you like having the target of the new authentications introduced in your mind. I was not my goal to keep the old way of authentication. I just wanted to add the bits necessary to make it work in our situation. I would appreciate if you could accept this for now because we need to go ahead with our Camunda based project. Of course I'm willing to support you, but our primary goal is to deliver our business software in time.

Reply:
@stephanpelikan I'm happy to work as well with you and really value your contributions. Looking forward to great collaborations ahead!

Reply:
@stephanpelikan `8.3.4.2` and `8.4.0-alpha2-rc3` has been released and should be publicly available shortly